benchmarks/transactions-generator/src/lib.rs

```diff
@@ -0,0 +1,217 @@
+use near_async::messaging::AsyncSender;
+use near_client::GetBlock;
+use near_client_primitives::types::GetBlockError;
+use near_network::client::{ProcessTxRequest, ProcessTxResponse};
+use near_primitives::hash::CryptoHash;
+use near_primitives::transaction::SignedTransaction;
+use near_primitives::types::BlockReference;
+use near_primitives::views::BlockView;
+use node_runtime::metrics::TRANSACTION_PROCESSED_FAILED_TOTAL;
+use node_runtime::metrics::TRANSACTION_PROCESSED_SUCCESSFULLY_TOTAL;
+use rand::distributions::{Distribution, Uniform};
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+use std::path::PathBuf;
+use std::time::Duration;
+use tokio::sync::oneshot;
+use tokio::task;
+
+pub mod account;
+#[cfg(feature = "with_actix")]
+pub mod actix_actor;
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub struct TxGeneratorConfig {
+    tps: u64,
+    volume: u64,
+    accounts_path: PathBuf,
+}
+
+impl Default for TxGeneratorConfig {
+    fn default() -> Self {
+        Self { tps: 0, volume: 0, accounts_path: "".into() }
+    }
+}
+
+#[derive(Clone, near_async::MultiSend, near_async::MultiSenderFrom)]
+pub struct ClientSender {
+    pub tx_request_sender: AsyncSender<ProcessTxRequest, ProcessTxResponse>,
+}
+
+#[derive(Clone, near_async::MultiSend, near_async::MultiSenderFrom)]
+pub struct ViewClientSender {
+    pub block_request_sender: AsyncSender<GetBlock, Result<BlockView, GetBlockError>>,
+}
+
+pub struct TxGenerator {
+    pub params: TxGeneratorConfig,
+    client_sender: ClientSender,
+    view_client_sender: ViewClientSender,
+    runner: Option<(task::JoinHandle<()>, oneshot::Sender<bool>)>,
+}
+
+#[derive(Debug, Clone)]
+struct Stats {
+    pool_accepted: u64,
+    pool_rejected: u64,
+    processed: u64,
+    failed: u64,
+}
+
+impl std::ops::Sub for Stats {
+    type Output = Self;
+    fn sub(self, other: Self) -> Self {
+        Self {
+            pool_accepted: self.pool_accepted - other.pool_accepted,
+            pool_rejected: self.pool_rejected - other.pool_rejected,
+            processed: self.processed - other.processed,
+            failed: self.failed - other.failed,
+        }
+    }
+}
+
+impl TxGenerator {
+    pub fn new(
+        params: TxGeneratorConfig,
+        client_sender: ClientSender,
+        view_client_sender: ViewClientSender,
+    ) -> anyhow::Result<Self> {
+        Ok(Self { params, client_sender, view_client_sender, runner: None })
+    }
+
+    pub fn start(self: &mut Self) -> anyhow::Result<()> {
+        if let Some(_) = self.runner {
```

Comment:
```suggestion
        if self.runner.is_some() {
```

Seems cleaner to me at least.